cvmfs/sync_item.h

```diff
@@ -169,6 +174,7 @@ class SyncItem {
   mutable EntryStat scratch_stat_;
 
   bool whiteout_;
```

Comment:
Added some additional documentation. Union file system concepts are discussed in more detail in `SyncUnion` which is referenced in `SyncItem` several times. I believe the attentive audience should find that. :-)